utopia-remix/app/routes/v1.projects.tsx

```diff
@@ -1,10 +1,15 @@
 import { LoaderFunctionArgs } from '@remix-run/node'
 import { handleListProjects } from '../handlers/listProjects'
 import { handle, handleOptions } from '../util/api.server'
+import { validateProjectAccess } from '../handlers/validators'
+import { UserProjectPermission } from '../types'
 
 export async function loader(args: LoaderFunctionArgs) {
   return handle(args, {
     OPTIONS: handleOptions,
-    GET: handleListProjects,
+    GET: {
+      handler: handleListProjects,
+      validator: validateProjectAccess(UserProjectPermission.CAN_VIEW_PROJECT),
```

Comment:
How is the validator working here when there aren't any params?

Reply:
defintiely right. It seems that this route indeed shouldn't work (shouldn't have used a validator at all, was a bad search/replace when added all of them), 
but looks like perhaps it is not actually used (since `/projects` leads to our new `projects.tsx` screen)?
in any case I reverted it back to `ALLOW`, since perhaps there is a place that calls it using `v1`.
I also added a way to pass the project param to the validator (according to Rheese's comment about `x.$projectId.ts` routes). We discussed it and perhaps the proper solution there will probably be to have another handler that extracts the project details before validation, but seems like it is out of scope of this "Project Visibility" PR. (but can defintiely be a subsequent "BFF middleware infra" PR).